inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSink.java

```diff
@@ -295,8 +298,10 @@ public void update() {
         if (zoneProducer == null) {
             return;
         }
+        reentrantLock.lock();
         lastNotifyTime.set(System.currentTimeMillis());
-        syncLock.notifyAll();
+        condition.signal();
+        reentrantLock.unlock();
```

Comment:
It will have problem if condition.signal() have an exception.
reentrantLock.unlock() need a finally block.

Reply:
done